src/VASSAL/counters/Stack.java

```diff
@@ -76,14 +78,25 @@ public Stack(GamePiece p) {
    * @return an Enumeration of the pieces in the stack, from the bottom up This
    *         is a clone of the contents so add/remove operations during read
    *         won't affect it.
-   * @deprecated
+   * @deprecated use {@link #asList()}
    */
   @Deprecated
   public Enumeration<GamePiece> getPieces() {
     return new EnumeratedIterator<>(new AllPieceIterator());
     //    return new AllPieceEnum();
   }
 
+  /**
+   * @return an unmodifiable {@link List} of {@link GamePiece}s contained in this {@link Stack}
+   */
+  public List<GamePiece> asList() {
+    List<GamePiece> result = new ArrayList<>();
+    for (int i = 0; i < pieceCount; i++) {
+      result.add(contents[i]);
+    }
+    return Collections.unmodifiableList(result);
+  }
+
```

Comment:
Oh! I just noticed that we're making a copy anyway, because `contents` is an array.

To avoid that, and also because it's much shorter, `asList()` could be:

```
return Collections.unmodifiableList(Arrays.asList(contents).sublist(0, pieceCount));
```

Reply:
Looks good